Mathlib/Order/SuccPred/Limit.lean

```diff
@@ -654,21 +637,99 @@ theorem isPredPrelimitRecOn_pred' (hb : ¬ IsMin b) :
     isPredPrelimitRecOn (pred b) hs hl = hs b hb :=
   isSuccPrelimitRecOn_succ' (α := αᵒᵈ) _ _ _
 
-@[deprecated isPredPrelimitRecOn_pred' (since := "2024-09-05")]
-alias isPredLimitRecOn_pred' := isPredPrelimitRecOn_pred'
-
 @[simp]
 theorem isPredPrelimitRecOn_pred [NoMinOrder α] (b : α) :
     isPredPrelimitRecOn (pred b) hs hl = hs b (not_isMin b) :=
   isPredPrelimitRecOn_pred' _ _ _
 
-@[deprecated isPredPrelimitRecOn_pred (since := "2024-09-05")]
-alias isPredLimitRecOn_pred := isPredPrelimitRecOn_pred
-
 end LinearOrder
 
 end isPredPrelimitRecOn
 
+section isSuccLimitRecOn
+
+section PartialOrder
+
+variable [PartialOrder α] [SuccOrder α]
+  (hm : ∀ a, IsMin a → C a) (hs : ∀ a, ¬ IsMax a → C (succ a)) (hl : ∀ a, IsSuccLimit a → C a)
+
+variable (b) in
+open Classical in
+/-- A value can be built by building it on minimal elements, successors, and successor limits. -/
+@[elab_as_elim]
+noncomputable def isSuccLimitRecOn : C b :=
+  isSuccPrelimitRecOn b hs fun a ha ↦
+    if h : IsMin a then hm a h else hl a (ha.isSuccLimit_of_not_isMin h)
+
+@[simp]
+theorem isSuccLimitRecOn_limit (hb : IsSuccLimit b) : isSuccLimitRecOn b hm hs hl = hl b hb := by
```

Comment:
Maybe take the opportunity to fix this name and `isSuccPrelimitRecOn_limit`?
```suggestion
theorem isSuccLimitRecOn_of_isSuccLimit (hb : IsSuccLimit b) : isSuccLimitRecOn b hm hs hl = hl b hb := by
```

Reply:
I don't like how wordy the new names are, but I guess they are more consistent.